util/vkgcElfReader.cpp

```diff
@@ -210,14 +212,24 @@ template <class Elf> unsigned ElfReader<Elf>::getRelocationCount() const {
 // @param idx : Relocation index
 // @param [out] reloc : Info of the relocation
 template <class Elf> void ElfReader<Elf>::getRelocation(unsigned idx, ElfReloc *reloc) const {
-  auto &section = m_sections[m_relocSecIdx];
-
-  auto relocs = reinterpret_cast<const typename Elf::Reloc *>(section->data);
-  reloc->offset = relocs[idx].r_offset;
-  reloc->symIdx = relocs[idx].r_symbol;
-  reloc->type = relocs[idx].r_type;
-  reloc->addend = 0;
-  reloc->useExplicitAddend = false;
+  auto *section = m_sections[m_relocSecIdx];
+
+  if (section->secHead.sh_type == SHT_REL) {
```

Comment:
For consistency I should probably have checked the section name here instead of the type. But I could not bear to add a string compare to this otherwise fast function.

Reply:
Moving in the right direction, even if it's by a little bit, is a good thing ;)